### PR TITLE
Keyboard Frame Notification

### PR DESCRIPTION
**Pull request by andersio**
Not quite sure about the name and where to place it though. `UIView.Type` seems a better place than `NotificationCenter`, since this is tied to the default notification center.

```swift
public struct KeyboardChangeContext {
	public var beginFrame: CGRect { get }
	public var endFrame: CGRect { get }
	public var animationCurve: UIViewAnimationCurve { get }
	public var animationDuration: Double { get }
	@available(iOS 9.0, *)
	public var isLocal: Bool { get }
}

extension Reactive where Base: UIView {
	public static var keyboardChange: Signal<KeyboardChangeContext, NoError>
}
```

### CODE REVIEW

File: ReactiveCocoa/UIKit/iOS/UIKeyboard.swift [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import UIKit
+import ReactiveSwift
+import enum Result.NoError
+
+public struct KeyboardChangeContext {
```
Comment (NachoSoto) [inline ReactiveCocoa/UIKit/iOS/UIKeyboard.swift]: Awesome. Can you add a quick docstring?

File: ReactiveCocoaTests/UIKit/UIKeyboardSpec.swift [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import ReactiveCocoa
+import Quick
+import Nimble
+
+class UIKeyboardSpec: QuickSpec {
+	override func spec() {
+		describe("UIView.Type.reactive.keyboardChange") {
+			it("should emit a `value` event when the notification is posted") {
+				var context: KeyboardChangeContext?
+
```
Comment (NachoSoto) [inline ReactiveCocoaTests/UIKit/UIKeyboardSpec.swift]: No need to give the parameter a name if it's going to be abbreviated :P just use `$0`?

File: ReactiveCocoaTests/UIKit/UIKeyboardSpec.swift [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+				}
+
+				let dummyInfo: [AnyHashable: Any] = [
+					UIKeyboardFrameBeginUserInfoKey: CGRect(x: 10, y: 10, width: 10, height: 10),
+					UIKeyboardFrameEndUserInfoKey: CGRect(x: 20, y: 20, width: 20, height: 20),
+					UIKeyboardAnimationDurationUserInfoKey: 1.0,
+					UIKeyboardAnimationCurveUserInfoKey: UIViewAnimationCurve.easeInOut
+				]
+
+				NotificationCenter.default.post(name: .UIKeyboardWillChangeFrame,
```
Comment (NachoSoto) [inline ReactiveCocoaTests/UIKit/UIKeyboardSpec.swift]: Using `NotificationCenter.default` could potentially affect other tests. Can you make the signal take a `NotificationCenter` (defaulting to this) so the test can use its own one?

**Comment by calebd** [comment]
Personally I don't think `UIView` is the appropriate place for this. There are plenty of notifications that are only posted in the default notification center but that doesn't mean they should be chained off of random classes in UIKit or Foundation.

**Comment by andersio** [comment]
Hmm, what about the `NotificationCenter` type?

File: ReactiveCocoaTests/UIKit/UIKeyboardSpec.swift (hunk 3 above)
Comment (andersio) [inline ReactiveCocoaTests/UIKit/UIKeyboardSpec.swift]: It would then have to be exposed as an extension to `NotificationCenter` instances, I'd think.

**Pull request merged**